### PR TITLE
Add the real code of accident percentage comparison per vehicle type

### PR DESCRIPTION
**Pull request by BarVolunteering**

### CODE REVIEW

**Comment by BarVolunteering** [comment]
Close #1222

File: anyway/infographics_utils.py [hunk 1]
```diff
@@ -490,12 +492,51 @@ def get_latest_accident_date(table_obj, filters):
...
+
+@lru_cache(maxsize=64)
+def percentage_accidents_by_car_type_national_data_cache(start_time, end_time):
+    involved_by_vehicle_type_data = get_accidents_stats(
+        table_obj=InvolvedMarkerView,
+        group_by="involve_vehicle_type",
+        count="involve_vehicle_type",
+        start_time=start_time,
+        end_time=end_time,
+    )
```
Comment (atalyaalon) [inline anyway/infographics_utils.py:527]: @BarVolunteering perhaps it's better to create an aggregated query on the national data, rather then extract all data and then perform aggregation?

**Review by atalyaalon — CHANGES_REQUESTED** [review]
@BarVolunteering great work - I added one comment, let me know what you think

**Comment by atalyaalon** [comment]
Also there is a very small conflict - I believe you can resolve it easily.

**Comment by BarVolunteering** [comment]
@atalyaalon I will take care of it

**Comment by idosavion** [comment]
Any chance we can add some test for the logic in this PR? Will be willing to help if needed.

**Comment by BarVolunteering** [comment]
@idosavion
I don't mind to add tests for the logic, but I might need some help in that
As this logic result is based on the Data in the DB I need to know what Data there is in GitHub automatic testing environment

**Comment by idosavion** [comment]
`percentage_accidents_by_car_type` doesn't rely on the db state, 
`stats_accidents_by_car_type_with_national_data` can be tested using [magicMock](https://docs.python.org/3/library/unittest.mock.html), I'm available on slack if you need any help

**Comment by BarVolunteering** [comment]
About the tests - I see that the relevant code section for the widget tests is comment out:
https://github.com/hasadna/anyway/blob/9396512535042bcbb4d0696ce56be260600a5d4f/tests/test_infographic_api.py#L85

Does anyone know why?

**Comment by atalyaalon** [comment]
> About the tests - I see that the relevant code section for the widget tests is comment out:
> https://github.com/hasadna/anyway/blob/9396512535042bcbb4d0696ce56be260600a5d4f/tests/test_infographic_api.py#L85
> 
> Does anyone know why?

After using git blame - it seems like @DavidShemian might know?

**Comment by idosavion** [comment]
We can just remove the commented out code, if you will need it in the future (and I doubt anyone is going to come back for it), he/she can find it in the git history.

File: anyway/infographics_utils.py [hunk 2]
```diff
@@ -490,12 +492,90 @@ def get_latest_accident_date(table_obj, filters):
...
+        elif vehicle_type in BE_CONST.LARGE_VEHICLE_TYPES:
+            driver_types["מסחרי/משאית"] += count
+        elif vehicle_type in BE_CONST.MOTORCYCLE_VEHICLE_TYPES:
+            driver_types["אופנוע"] += count
+        elif vehicle_type in BE_CONST.BICYCLE_AND_SMALL_MOTOR_VEHICLE_TYPES:
+            driver_types["אופניים/קורקינט"] += count
+        else:
+            driver_types["אחר"] += count
+
+    # So we have a problem here, when we Calculate percentage and convert the value to int - we are losing some of the
```
[the remark this answers is not in the document]
Comment (BarVolunteering) [reply anyway/infographics_utils.py:512]: I got the impression to return int form the google docs info, I will change it to float

**Review by atalyaalon — APPROVED** [review]
@BarVolunteering great work!

File: tests/test_infographic_api.py [hunk 3]
```diff
@@ -75,6 +79,57 @@ def test_bad_news_flash_id(self, app):
...
         assert rv.status_code == http_client.NOT_FOUND
 
+    def test_accident_count_by_car_type(self, app):
+        test_involved_by_vehicle_type_data = [{"involve_vehicle_type": 1, "count": 3}, {"involve_vehicle_type": 25, "count": 2},
+         {"involve_vehicle_type": 15, "count": 1}]
+        output_tmp = percentage_accidents_by_car_type(test_involved_by_vehicle_type_data)
+        assert len(output_tmp) == 3
+        assert output_tmp["רכב פרטי"] == 50
+        assert output_tmp["מסחרי/משאית"] == 33.333333333333336
+        assert output_tmp["אופניים/קורקינט"] == 16.666666666666668
```
Comment (idosavion) [inline tests/test_infographic_api.py:89]: sorry, one last thing.. consider using  self.assertAlmostEqual()
thanks for adding the tests!

File: tests/test_infographic_api.py [hunk 4]
```diff
@@ -75,6 +79,57 @@ def test_bad_news_flash_id(self, app):
...
+                 {'car_type': 'אופנוע',
+                  'percentage_country': 6.262912323870099,
+                  'percentage_segment': 0.0},
+                 {'car_type': 'אופניים/קורקינט',
+                  'percentage_country': 2.794865247568421,
+                  'percentage_segment': 0.0}]
+
+        infographics_utils.get_accidents_stats = tmp_func  # Restore function ref - So we don't affect other tests
+        assert len(out) == len(good)
+        assert out == good
```
Comment (idosavion) [inline tests/test_infographic_api.py:130]: expected & actual might be more suitable IMO, not mandatory

File: tests/test_infographic_api.py (hunk 3 above)
Comment (BarVolunteering) [reply tests/test_infographic_api.py:89]: assertAlmostEqual is python default unittest library function, we are using pytest - the best similarity I found is pytest.approx

**Review by idosavion — APPROVED** [review]
Great PR!

**Comment by idosavion** [comment]
LGTM, can you squash the commits before merging?

**Comment by BarVolunteering** [comment]
@idosavion Yes, its part of the merge system option:
image

**Comment by idosavion** [comment]
Not a good way to merge, look at the commit message

**Comment by BarVolunteering** [comment]
@idosavion The commit message is changeable during the merge process, I saw from a number of merge squashs(e.g [Merge 1](https://github.com/hasadna/anyway/commit/9396512535042bcbb4d0696ce56be260600a5d4f), [Merge 2](https://github.com/hasadna/anyway/commit/30971d9c93d8d9310eeeb9abada31bfec8c6f39e), [Merge 3](https://github.com/hasadna/anyway/commit/0276bc97fd5b3a894e91b88fe422860818aafc8d) and more) that this is standard way to log merge squashes in this project

**Comment by idosavion** [comment]
when the commits are meaningful, It's a good way to do it, all the commit messages you provided above are bad IMO.

**Comment by BarVolunteering** [comment]
I understood that all commit messages are needed to be copied to the merge squash message but I have no problem with changing the message in future* merge squashes.

*Changing the current message in this merge will delete the merge commit and create a new one - and there is the possibility that someone already fetched this repo and so he will have problems when pushing his/her work.

**Comment by idosavion** [comment]
Of course :)

**Pull request merged**